src/lcm/interfaces.py

```diff
@@ -6,6 +7,7 @@
 
 from lcm.grids import ContinuousGrid, DiscreteGrid, Grid
 from lcm.typing import InternalUserFunction, ParamsDict, ShockType
+from lcm.utils import first_non_none
 
 
 @dataclass(frozen=True)
```

Comment:
Does this make the mutable attributes frozen, too? Else, would there be a value in using `FrozenDicts` or the like?

Reply:
I does not!

I am unsure whether `FrozenDicts` have a net-positive value. I'd wait until we have more usage data before adding another dependency.

Reply:
I guess it would add safety. From searching and comparing a bit, [immutables.Map](https://github.com/MagicStack/immutables) seems like they would be the way to go in case we decide for that.